pkg/faketikv/cluster.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package faketikv
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/pd/server/core"
+)
+
+// ClusterInfo records all cluster information
+type ClusterInfo struct {
+	*core.RegionsInfo
+	Nodes       map[uint64]*Node
+	firstRegion *core.RegionInfo
+}
+
+// GetBootstrapInfo returns first region and it's leader store
+func (c *ClusterInfo) GetBootstrapInfo() (*metapb.Store, *metapb.Region) {
+	storeID := c.firstRegion.Leader.GetStoreId()
```

Comment:
Is it possible that the region has no leader?

Reply:
the Initializer should ensure it.